### ISSUE TITLE
Clean build gpu-pytorch breaks Databricks notebooks environment

### ISSUE THREAD

**Issue by iiLaurens** [databricks/containers#36]
I have a pipeline which is adapted from the gpu-pytorch Dockerfile provided in this repository. However somewhere this week the conda environment must choose a different combination of packages in its environment, since new clean images build from this Dockerfile no longer run on Databricks. The problem is that any executed command in a notebook will have a _Waiting to run..._ status and then gets cancelled after a minute. The driver logs seem to mention this problem:

```
Traceback (most recent call last):
  File "/local_disk0/tmp/1600798981779-0/PythonShell.py", line 1732, in <module>
    raise e
  File "/local_disk0/tmp/1600798981779-0/PythonShell.py", line 1729, in <module>
    launch_process()
  File "/local_disk0/tmp/1600798981779-0/PythonShell.py", line 1716, in launch_process
    console_buffer, error_buffer)
  File "/local_disk0/tmp/1600798981779-0/PythonShell.py", line 783, in __init__
    self.shell = self.create_shell()
  File "/local_disk0/tmp/1600798981779-0/PythonShell.py", line 1153, in create_shell
    autocomplete_verbose_output=self.autocomplete_verbose_output()
  File "/local_disk0/pythonVirtualEnvDirs/virtualEnv-8f2ea4ca-f011-4eb0-8d56-13e442652e87/lib/python3.7/site-packages/traitlets/config/configurable.py", line 510, in instance
    inst = cls(*args, **kwargs)
  File "/local_disk0/pythonVirtualEnvDirs/virtualEnv-8f2ea4ca-f011-4eb0-8d56-13e442652e87/lib/python3.7/site-packages/IPython/terminal/embed.py", line 159, in __init__
    super(InteractiveShellEmbed,self).__init__(**kw)
  File "/local_disk0/pythonVirtualEnvDirs/virtualEnv-8f2ea4ca-f011-4eb0-8d56-13e442652e87/lib/python3.7/site-packages/IPython/terminal/interactiveshell.py", line 519, in __init__
    super(TerminalInteractiveShell, self).__init__(*args, **kwargs)
  File "/local_disk0/pythonVirtualEnvDirs/virtualEnv-8f2ea4ca-f011-4eb0-8d56-13e442652e87/lib/python3.7/site-packages/IPython/core/interactiveshell.py", line 676, in __init__
    self.init_completer()
  File "/local_disk0/tmp/1600798981779-0/PythonShell.py", line 1003, in init_completer
    if self.use_jedi:
AttributeError: 'IPythonShell' object has no attribute 'use_jedi'
```

Strangely enough the latest images of `databricksruntime/gpu-pytorch:cuda10.1` on Dockerhub (build 3 months) ago still seem to run fine in Databricks. I like to append extra packages to the existing to the conda environment, so using `FROM databricksruntime/gpu-pytorch:cuda10.1` in my Dockerfile is not really preferred. Also I was able to build and run my images succesfully up until now. Something has changed in how conda resolves the environment upon installation that breaks compatibility with databricks notebooks. Anyone knows what the issue might be?

**Comment by iiLaurens** [comment]
I did some more detective work to find out the differences in the conda environment. Have a look at the diffs in this gist:
[https://gist.github.com/iiLaurens/ba794b40cd3cde72fdf52d2ebf05a95c/revisions](https://gist.github.com/iiLaurens/ba794b40cd3cde72fdf52d2ebf05a95c/revisions)

It shows the results from the `conda list` command in the image currently (https://hub.docker.com/layers/databricksruntime/gpu-pytorch/cuda10.1/images/sha256-14d119bdbe6f07919952d60b54153430a474664c786c60b26634638039fa0bee?context=explore)[hosted on Dockerhub] compared to the `conda list` of the image that I newly build, based of the (https://github.com/databricks/containers/blob/master/ubuntu/gpu/pytorch/Dockerfile)[Dockerfile in this repository].

**Comment by thompdre84** [comment]
I had this error all day yesterday. The issue was Traitlets. Make sure to specify the correct version of traitlets in your env.yml file. The one that gets installed by default into the conda environment (5.x.x) is what breaks everything.

**Comment by iiLaurens** [comment]
> I had this error all day yesterday. The issue was Traitlets. Make sure to specify the correct version of traitlets in your env.yml file. The one that gets installed by default into the conda environment (5.x.x) is what breaks everything.
> 
> edit: In my case I was using the runtime 7 Dockerfile example that is found under open pull requests. It was working great last week, then Monday morning everything suddenly stopped with the "jedi" error. I set traitlets=4.3.3 (the 7.2 runtime version) in env.yml and all is well again.

That's some serious investigative work to figure that one out! Thanks for sharing. Is traitlets installed in all ipython environments, such that all of the docker images in this repo are affected?

**Comment by thompdre84** [comment]
> That's some serious investigative work to figure that one out! Thanks for sharing. Is traitlets installed in all ipython environments, such that all of the docker images in this repo are affected?

I believe traitlets is installed automatically when ipython is installed (in the conda environment) because this was my whole env.yml before the error issue started: 
`name: dcs-minimal`
`channels:`
`  - default`
`dependencies:`
`  - pip:`
`    - pyarrow==0.15.1`
`  - python=3.7.5`
`  - six=1.14.0`
`  - nomkl=3`
`  - numpy=1.18.1`
`  - pandas=1.0.1`
`  - ipython=7.12.0`
`  - jedi=0.14.1`

**Comment by mengxr** [comment]
Thanks for reporting the issue and the investigation! We merged #37 to pin traitlets version to fix the issue.
